frontend/src/pages/editProfile.js

```diff
@@ -1,40 +1,91 @@
-import React from "react";
-import { useSelector, useDispatch } from "react-redux";
+import React, { useEffect, useState } from "react";
 import { useNavigate } from "react-router-dom";
 import Button from "react-bootstrap/Button";
 import Form from "react-bootstrap/Form";
 import Container from "react-bootstrap/Container";
 import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
 
-import { editProfile } from "../store/profile";
 import InputField from "../components/input";
-import { validate, validateField, getErrorPayload } from "../utils/validation";
+import { validate, validateField } from "../utils/validation";
+import * as profileService from "../services/profileService";
+import { getErrorMessage } from "../utils/validation";
 
 const EditProfile = () => {
-  const state = useSelector((state) => state.editProfile.value);
-  const dispatch = useDispatch();
+  const [state, setState] = useState({ errors: {} });
+
   const navigate = useNavigate();
+  let avatarFile = {};
+
+  useEffect(() => {
+    const fetchProfile = async () => {
+      const { data } = await profileService.getCurrentProfile();
+      const profile = data.profile;
+      delete profile["avatar"];
+      delete profile["total_words_learned"];
+      delete profile["total_lessons_learned"];
+      const user = data.user;
+      const errors = {};
+      const payload = { ...state, ...profile, ...user, errors };
+
+      setState(payload);
```

Comment:
1. separate the state of errors so you dont need to set errors here which is not necessary
2. just set all response, no need to delete not needed data, you can just ignore or remove those unnecessary data from backend before passing to FE

```suggestion
      const { data } = await profileService.getCurrentProfile();
      setState(data);
```